File: python_cpp/uPIMulator_frontend/src/abi/isa/cc/true_cc.py
from typing import Set

from abi.isa.cc._base_cc import BaseCC
from abi.isa.instruction.condition import Condition


class TrueCC(BaseCC):
    def __init__(self, condition: Condition):
        super().__init__(TrueCC.conditions(), condition)

    @staticmethod
    def conditions() -> Set[Condition]:
        return {Condition.TRUE}

File: golang_vm/uPIMulator/src/device/linker/kernel/instruction/cc/true_cc.go
package cc

import (
	"errors"
)

type TrueCc struct {
	condition Condition
}

func (this *TrueCc) Init(condition Condition) {
	conditions := map[Condition]bool{
		TRUE: true,
	}

	if _, found := conditions[condition]; !found {
		err := errors.New("condition is not allowed")
		panic(err)
	}

	this.condition = condition
}

func (this *TrueCc) Condition() Condition {
	return this.condition
}
